App shows the setup screen when already setup

# Bug Report


## Problem 

Sometimes, the app shows the setup screen despite being already setup.

Not sure what makes it happens, I would chat with someone, then go to another app and back, and I get the setup screen.

If I force kill the app in android, and reopen it, it goes back to normal and prompt password for my profile.

![Screenshot_20240731-140530](https://github.com/user-attachments/assets/e345e72a-2a59-45fb-9549-5ccf33a2a56d)


### Additional Information

- Status version: 97b19c54401b9fffc003f0b79feff83e21f55851 
- Operating System: Android, Pixel 7

Happened again. This time I only have the play store version installed for a while.

```
type: logcat
osVersion: google/panther/panther:14/AP2A.240905.003/2024092900:user/release-keys
packageName: im.status.ethereum:2024082617
buffers: main,system,crash,events,kernel
level: verbose

--------- beginning of crash
10-10 08:18:03.791  3306  3306 E AndroidRuntime: FATAL EXCEPTION: main
10-10 08:18:03.791  3306  3306 E AndroidRuntime: Process: im.status.ethereum, PID: 3306
10-10 08:18:03.791  3306  3306 E AndroidRuntime: java.lang.RuntimeException: Unable to start service im.status.ethereum.pushnotifications.ForegroundService@c65e810 with Intent { cmp=im.status.ethereum/.pushnotifications.ForegroundService }: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service im.status.ethereum/.pushnotifications.ForegroundService
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5126)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2439)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8623)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: Caused by: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service im.status.ethereum/.pushnotifications.ForegroundService
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Parcel.readParcelableInternal(Parcel.java:5036)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Parcel.readParcelable(Parcel.java:5018)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3187)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3176)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3152)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3094)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6998)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:879)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at im.status.ethereum.pushnotifications.ForegroundService$$ExternalSyntheticApiModelOutline2.m(SourceFile:0)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at im.status.ethereum.pushnotifications.ForegroundService.onStartCommand(Unknown Source:225)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5108)
10-10 08:18:03.791  3306  3306 E AndroidRuntime: 	... 11 more
10-10 10:05:43.583 17431 17431 E AndroidRuntime: FATAL EXCEPTION: main
10-10 10:05:43.583 17431 17431 E AndroidRuntime: Process: im.status.ethereum, PID: 17431
10-10 10:05:43.583 17431 17431 E AndroidRuntime: java.lang.RuntimeException: Unable to start service im.status.ethereum.pushnotifications.ForegroundService@e9037d5 with Intent { cmp=im.status.ethereum/.pushnotifications.ForegroundService }: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service im.status.ethereum/.pushnotifications.ForegroundService
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5126)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2439)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8623)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: Caused by: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service im.status.ethereum/.pushnotifications.ForegroundService
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Parcel.readParcelableInternal(Parcel.java:5036)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Parcel.readParcelable(Parcel.java:5018)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3187)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3176)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3152)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3094)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6998)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:879)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at im.status.ethereum.pushnotifications.ForegroundService$$ExternalSyntheticApiModelOutline2.m(SourceFile:0)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at im.status.ethereum.pushnotifications.ForegroundService.onStartCommand(Unknown Source:225)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5108)
10-10 10:05:43.583 17431 17431 E AndroidRuntime: 	... 11 more
--------- beginning of events
10-15 08:42:26.879 25240 25240 I wm_on_restart_called: [Token=40512448,Component Name=im.status.ethereum.MainActivity,Reason=performRestart,time=0ms]
10-15 08:42:26.890 25240 25240 I wm_on_start_called: [Token=40512448,Component Name=im.status.ethereum.MainActivity,Reason=handleStartActivity,time=10ms]
--------- beginning of main
10-15 08:42:26.891 25240 25240 W WindowOnBackDispatcher: OnBackInvokedCallback is not enabled for the application.
10-15 08:42:26.891 25240 25240 W WindowOnBackDispatcher: Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
--------- switch to events
10-15 08:42:26.935 25240 25240 I wm_on_resume_called: [Token=40512448,Component Name=im.status.ethereum.MainActivity,Reason=RESUME_ACTIVITY,time=32ms]
10-15 08:42:26.935 25240 25240 I wm_on_top_resumed_gained_called: [Token=40512448,Component Name=im.status.ethereum.MainActivity,Reason=topWhenResuming]
--------- switch to main
10-15 08:42:26.949 25240 25780 D ReactNativeJS: 2024-10-15T08:42:26.943Z DEBUG [native-module.core:330] - [native-module] app-state-change
10-15 08:42:26.951 25240 25780 D ReactNativeJS: 2024-10-15T08:42:26.950Z DEBUG [status-im.common.keychain.events:35] - [keychain] can-save-user-password?
10-15 08:42:26.952 25240 25781 D StatusModule: AppStateChange: active
--------- switch to events
10-15 08:42:26.978 25240 25240 I viewroot_draw_event: [window=VRI[MainActivity],event=Start draw after previous draw not visible]
--------- switch to main
10-15 08:42:26.995 25240 25530 E GoLog   : WARN [10-15|08:42:26.972] messenger nil, not reporting app state change
--------- switch to events
10-15 08:42:26.995 25240 25240 I viewroot_draw_event: [window=VRI[MainActivity],event=reportDrawFinished seqId=0]
--------- switch to main
10-15 08:42:26.996 25240 25781 D RNKeychainManager: Probe cipher storage: CipherStorageFacebookConceal
10-15 08:42:26.997 25240 25781 D RNKeychainManager: Probe cipher storage: CipherStorageKeystoreAesCbc
10-15 08:42:26.997 25240 25781 D RNKeychainManager: Probe cipher storage: CipherStorageKeystoreRsaEcb
10-15 08:42:26.997 25240 25781 D RNKeychainManager: Selected storage: CipherStorageKeystoreAesCbc
10-15 08:42:26.998 25240 25780 D ReactNativeJS: 2024-10-15T08:42:26.997Z DEBUG [native-module.core:375] - [native-module] rooted-device?
--------- switch to events
10-15 08:42:27.504 25240 25240 I wm_on_top_resumed_lost_called: [Token=40512448,Component Name=im.status.ethereum.MainActivity,Reason=topStateChangedWhenResumed]
10-15 08:42:27.560 25240 25240 I view_enqueue_input_event: [eventType=Motion - Cancel,action=im.status.ethereum/im.status.ethereum.MainActivity]
10-15 08:42:29.438 25240 25240 I wm_on_paused_called: [Token=40512448,Component Name=im.status.ethereum.MainActivity,Reason=performPause,time=5ms]
--------- switch to main
10-15 08:42:29.446 25240 25780 D ReactNativeJS: 2024-10-15T08:42:29.441Z DEBUG [native-module.core:330] - [native-module] app-state-change
10-15 08:42:29.446 25240 25781 D StatusModule: AppStateChange: background
10-15 08:42:29.447 25240 25530 E GoLog   : WARN [10-15|08:42:29.446] messenger nil, not reporting app state change
10-15 08:42:29.823 25240 25240 D VRI[MainActivity]: visibilityChanged oldVisibility=true newVisibility=false
--------- switch to events
10-15 08:42:29.841 25240 25240 I viewroot_draw_event: [window=VRI[MainActivity],event=Not drawing due to not visible]
10-15 08:42:30.859 25240 25240 I wm_on_stop_called: [Token=40512448,Component Name=im.status.ethereum.MainActivity,Reason=STOP_ACTIVITY_ITEM,time=0ms]
```

--- a/modules/react-native-status/android/src/main/java/im/status/ethereum/pushnotifications/PushNotificationHelper.java
+++ b/modules/react-native-status/android/src/main/java/im/status/ethereum/pushnotifications/PushNotificationHelper.java
@@ -873,7 +873,7 @@ public class PushNotificationHelper {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
             context.startForegroundService(serviceIntent);
         } else {
-            context.startService (serviceIntent);
+            context.startService(serviceIntent);
         }
         this.registerBroadcastReceiver();
     }

--- a/modules/react-native-status/android/src/main/java/im/status/ethereum/pushnotifications/PushNotificationHelper.java
+++ b/modules/react-native-status/android/src/main/java/im/status/ethereum/pushnotifications/PushNotificationHelper.java
@@ -870,7 +870,11 @@ public class PushNotificationHelper {
     public void start() {
         Log.e(LOG_TAG, "Starting Foreground Service");
         Intent serviceIntent = new Intent(context, ForegroundService.class);
-        context.startService(serviceIntent);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            context.startForegroundService(serviceIntent);
+        } else {
+            context.startService (serviceIntent);
+        }
         this.registerBroadcastReceiver();
     }
 

